### PR TITLE
Updated to support Arris SB6183 and refactored code

### PR DESCRIPTION
**Pull request by jdburton**
This pull request includes the following changes:

- Added support for the Arris SB6183 
(Derived from https://github.com/billimek/SB6183-stats-for-influxdb/blob/master/SB6183.py) 

- Added support for modem model in config.py

- Added support for the Printer outputter as a config.py option

- Refactored code so that all modem models are subclasses of ArrisModem

### CODE REVIEW

File: scraper/targets/arris_modem_SB6183.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            modulation = table_row.find_all('td')[2].text.strip()
+            channel_id = table_row.find_all('td')[3].text.strip()
+            frequency = table_row.find_all('td')[4].text.replace(" Hz", "").strip()
+            power = table_row.find_all('td')[5].text.replace(" dBmV", "").strip()
+            snr = table_row.find_all('td')[6].text.replace(" dB", "").strip()
+            corrected = table_row.find_all('td')[7].text.strip()
+            uncorrectables = table_row.find_all('td')[8].text.strip()
+            octets = "0"
+
+    #        print("Channel = %s, Channel ID = %s, Frequency = %s, Power = %s, SNR = %s, corrected = %s, uncorrectables = %s" % (channel, channel_id, frequency, power, snr, corrected, uncorrectables))
```
Comment (twstokes) [inline scraper/targets/arris_modem_SB6183.py]:
```suggestion
```

File: scraper/targets/arris_modem_SB6183.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            'power',
+            'snr',
+            'correcteds',
+            'uncorrectables',
+            'octets',
+        ]
+        items = []
+        
+        for table_row in soup.find_all("table")[2].find_all("tr")[2:]:
+    #       print(table_row.find_all('td'))
```
Comment (twstokes) [inline scraper/targets/arris_modem_SB6183.py]:
```suggestion
```

File: scraper/targets/arris_modem_CM820A.py [hunk 3]
```diff
@@ -5,151 +5,91 @@
...
-        items.append(UpstreamItem(zipped.items()))
-
-    return items
+    #def extract_items_from_html(self, html_string):
+        #get items from the downstream table
+        #downstream_items = get_downstream_items(html_string)
+        #get items from the upstream table
+        #upstream_items = get_upstream_items(html_string)
+
+        #return downstream_items + upstream_items
```
Comment (twstokes) [inline scraper/targets/arris_modem_CM820A.py]:
```suggestion
```

File: config_sample.py [hunk 4]
```diff
@@ -2,9 +2,11 @@
 Parameters for the modem scraper.
 """
 scraper_config = {
+    'modem_model' '', #POPULATE MODEM MODEL HERE
```
Comment (twstokes) [inline config_sample.py]:
```suggestion
    'modem_model' '', # POPULATE MODEM MODEL HERE
```

File: scraper/targets/arris_modem_SB6183.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    Args:
+        Target (string): [HTML]
+    """
+    #def extract_items_from_html(self, html_string):
+        #get items from the downstream table
+        #downstream_items = get_downstream_items(html_string)
+        #get items from the upstream table
+        #upstream_items = get_upstream_items(html_string)
+
+        #return downstream_items + upstream_items
```
Comment (twstokes) [inline scraper/targets/arris_modem_SB6183.py]:
```suggestion
    """
    ArrisModem subclass that represents an Arris modem model SB6183
    running software 
    """
```

File: scraper/targets/arris_modem_TM1602AP2.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    """
+
+    #def extract_items_from_html(self, html_string):
+        #get items from the downstream table
+        #downstream_items = get_downstream_items(html_string)
+        #get items from the upstream table
+        #upstream_items = get_upstream_items(html_string)
+
+        #return downstream_items + upstream_items
+
```
Comment (twstokes) [inline scraper/targets/arris_modem_TM1602AP2.py:26]:
```suggestion
    """
    ArrisModem subclass that represents an Arris modem model TM1602AP2
    running software 9.1.103J6J.
    """

```

File: scraper/targets/arris_modem_CM820A.py [hunk 7]
```diff
@@ -5,151 +5,91 @@
...
 
-class ArrisModem(Target):
+class ArrisModemCM820A(ArrisModem):
     """
     Target subclass that represents an Arris modem model CM820A
     running software 9.1.103S.
 
     Args:
         Target (string): [HTML]
     """
```
Comment (twstokes) [inline scraper/targets/arris_modem_CM820A.py]:
```suggestion
    """
    ArrisModem subclass that represents an Arris modem model CM820A
    running software 9.1.103S.
    """
```

**Review by twstokes — APPROVED** [review]
Hey @jdburton 👋! Thanks for your additions, they're awesome!

I added some cleanup suggestions to the PR, no blockers, so feel free to merge and I can clean them up later. If any of them need tweaking feel free to do so.

I'll tag a new release after this gets merged to `master`.

**Comment by jdburton** [comment]
I cleaned up the code per your suggestions.

**Comment by twstokes** [comment]
Awesome! Thanks again @jdburton.

**Pull request merged**